gravitee-apim-cypress/cypress/integration/rest-api/apis/api-flow.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ADMIN_USER, API_PUBLISHER_USER, LOW_PERMISSION_USER } from 'fixtures/fakers/users/users';
+import { ApiFakers } from 'fixtures/fakers/apis';
+import { PlanFakers } from 'fixtures/fakers/plans';
+import {
+  createApi,
+  deleteApi,
+  deployApi,
+  publishApi,
+  startApi,
+  createPlan,
+  publishPlan,
+  deletePlan,
+  stopApi,
+} from 'commands/management/api-management-commands';
+import { Api } from 'model/apis';
+import { NewPlanEntity, SecurityType } from 'model/plan';
+
+context('Create an API flow', () => {
+  let api: Api;
+  let plan: NewPlanEntity;
+
+  describe('Create an API', function () {
+    it('should create an API as admin user', function () {
+      const fakeApi: Api = ApiFakers.api();
+      createApi(ADMIN_USER, fakeApi).should((response) => {
+        expect(response.body.state).equal('STOPPED');
+        expect(response.body.visibility).equal('PRIVATE');
+        expect(response.body.lifecycle_state).equal('CREATED');
+        api = response.body;
+      });
+    });
+
+    it('should fail to create an API if user lacks required permssions', function () {
+      const fakeApi: Api = ApiFakers.api();
+      createApi(LOW_PERMISSION_USER, fakeApi).should((response) => {
+        expect(response.status).to.equal(403);
+        expect(response.body.message).to.equal('You do not have sufficient rights to access this resource');
+      });
+    });
```

Comment:
I don't think it's really readable
What about having a `Context` "without problem", a `describe` representing a scenario and each `it` for each step of this scenario ?

And a second context for  error cases ? Or even another file ?

It's just an example. But I think it is easier to read when it's like a story.
It's not a change requirement, just an opinion. What do you think ?

Reply:
context() is exactly same as describe() but it dose improve readability little bit. We have added to this point and will implement in future. 

This test is designed as one user journey and at the moment not big enough to devide into two so we will leave it in single file. The disadvantage of  of adding all the negative scenarios to separate file is we will have to run all the prerequisites before we run the negative scenarios, this will increase run time.

Reply:
Yes, very good point, as usual :) And it confirms my feeling about it. 
I'll leave it like this for now and as a very next step we'll discuss QA internally how a cleaner solution should look like that can also cope with more failing tests.

Reply:
@patelami yes I agree Context and Describe are the same. 
It was more a remark about user journey indeed, I think mixing positive and negative cases add some complexity.
but no problem to me to keep it like that for the moment and make it evolve when it will grow